addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,227 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+from datetime import timedelta
+import re
+import requests
+from requests.exceptions import ConnectTimeout, HTTPError, ReadTimeout
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.misc import format_datetime
+
+class ProductGetPicUsingBarcode(models.TransientModel):
+    """ Get product pictures using barcode on Google Image
+
+    """
+    _name = 'product.images.get.pic.barcode'
+    _description = 'Get Picture with Barcode Number'
+
+    @api.model
+    def default_get(self, fields_list):
+        """ Override of default_get(fields_list) -> default_values
+
+        Return default values for the fields in ``fields_list``. Default
+        values are determined by the context, user defaults, and the model
+        itself.
+
+        :param list fields_list: names of field whose default is requested
+        :raises UserError: If a scheduled task is already triggered
+        :raises UserError: If the user didn't enter API Key or Search Engine ID in the settings
+        :raises UserError: If there isn't any product without a picture and with a barcode selected.
+        :return: a dictionary mapping field names to their corresponding default values,
+            if they have a default value.
+        :rtype: dict
+        """
+        # Check if the cron is already trigger and raise an appropriate UserError
+        cron_id = self.env.ref('product_images.ir_cron_get_pics_from_google').id
+        date = self.env['ir.cron.trigger'].search([('cron_id', '=', cron_id)]).call_at
+        if date:
+            raise UserError(_("Another task is already scheduled to pick the images on Google "
+                              "on %s", format_datetime(self.env, date)))
```

Comment:
It shouldn't be the case since we defined 9999 months as the interval in the ir_cron_data BUT even if it was, I'm looking for ir_cron_trigger: https://github.com/odoo/odoo/blob/8b45cce6897e058e1a2e42cb0ad324c5acb9575f/addons/product_images/wizard/product_get_pic_using_barcode.py#L38 
The only way to trigger it is either by the user or if there is an exception that triggers the cron later. Again, it shouldn't be the case. 
However, to avoid an UserError in the case where two or more triggers would be returned, we could add an indice:
` date = self.env['ir.cron.trigger'].search([('cron_id', '=', cron_id)])[0].call_at `

Reply:
I should have said "what if the cron triggers itself" *

What I mean is that a large number of images may take several days to be fetched. However, this message tells the user that the job will be finished today... every day, while it could take another couple of days to finish.

Reply:
Indeed, I didn't think of this case. I'll see with @pierre-pde to change the message.